File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    /*
     [1,2,3,3,4,4,5]
                ^ ^
     125
    
    */ 
      ListNode* deleteDuplicates(ListNode* head) {
        if(!head || !head->next) return head;
        ListNode *next = head->next;
        
        if(next->val == head->val){
            while(next && next->val == head->val) next=next->next;//skip duplicates
            return deleteDuplicates(next);//delete duplicate elements
        }else{
            head->next = deleteDuplicates(next);
            return head;
        }
    }
};

File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(-1)
        dummy.next = head
        prev = dummy
        curr = head
        
        while curr:
            duplicate = True
            while curr.next and curr.val == curr.next.val:
                curr = curr.next
                duplicate = False
            if duplicate:
                prev.next = curr
                prev = prev.next
            
            if curr:
                curr= curr.next
        prev.next = curr
        return dummy.next